x-pack/plugins/apm/public/utils/formatters.js

```diff
@@ -52,3 +52,14 @@ export function asInteger(value) {
 export function tpmUnit(type) {
   return type === 'request' ? 'rpm' : 'tpm';
 }
+
+export function getDurationPercent(
```

Comment:
To make it more generic and aligned with the other names in formatters we could call it `asPercent`, and use `numerator` and `denominator`:
```tsx
export function asPercent(
  numerator: number,
  denominator?: number,
  fallbackResult = ''
) {
  if (!denominator) {
    return fallbackResult;
  }
  return numeral(numerator / denominator).format('0.00%') as string;
}

```

Reply:
... and in combination with a `safeDivide` function:

```ts
export function safeDivide(numerator: number, denominator?: number) {
  if (!denominator) {
    return;
  }
  return numerator / denominator;
}

export function asPercent(
  numerator: number,
  denominator?: number,
  fallbackResult = ''
) {
  const res = safeDivide(numerator, denominator);
  if (!res) {
    return fallbackResult;
  }
  return numeral(res).format('0.00%') as string;
}
```

Reply:
I think it might make more sense if `safeDivide` just returns `undefined` as fallback, and then the caller is free to handle that however they want.
`asPercent` on the other hand takes care of formatting, and fallback for the consumer.

Reply:
Right now it's a little hard to anticipate all the use-cases. So probably better to start applying it in the codebase, and then we can see if it fits.

Reply:
Yeah, this lines up with what I was imagining.